docs/fundamentals/configuration.rst

```diff
@@ -1,6 +1,152 @@
 Configuration
 =============
+By :ref:`Steve Smith <configuration-author>` | Originally Published: 6 May 2015
 
-.. note::
+ASP.NET 5 supports a variety of different configuration options. Application configuration data can come from files using built-in support for JSON, XML, and INI formats, as well as from environment variables. Of course, developers can build their own custom configuration providers, as well.
 
-    This topic hasn't been written yet! You can track the status of this `issue <https://github.com/aspnet/Docs/issues/64>`_ through our public GitHub issue tracker. Learn how you can `contribute <https://github.com/aspnet/Docs/blob/master/CONTRIBUTING.md>`_ on GitHub.
+In this article:
+	- `Getting and setting configuration settings`_
+	- `Using the built-in providers`_
+	- `Writing custom providers`_
+	
+`Download sample from GitHub <https://github.com/aspnet/docs>`_.
+
+Getting and setting configuration settings
+------------------------------------------
+
+ASP.NET 5's configuration system has been re-architected from previous versions of ASP.NET, which relied on ``System.Configuration`` and XML configuration files like ``web.config``. The new `configuration model <https://github.com/aspnet/Configuration>`_ provides streamlined access to key/value based settings that can be retrieved from a variety of sources.
+
+To work with settings in your ASP.NET application, you can instantiate a new instance of ``Configuration`` anywhere you need one. However, it's recommended to do this once in your application's ``Startup``, and then inject an instance of ``IConfiguration`` into any controllers or services that need to access configuration. At its simplest, the ``Configuration`` class is just a collection of ``Sources``, which provide the ability to read and write name/value pairs. You must configure at least one source in order for ``Configuration`` to function correctly. For instance, you could include the following code in any method in your ASP.NET application:
```

Comment:
Are we going to change this to not tell people to inject configuration directly? The path we want to direct people down is using `Configuration` in their Startup.cs and then only using a poco application specific settings class(es) in their app.